File: react-frontend/src/Api/history.ts
import backend from "./backend";
import config from "../Shared/config";
//THis function will get the history of translation from flask-server if running on localhost or from localstorage if running on live
export const getHistory = async () => {
  let result: Array<TranslationHistory> = [];
  if (config.backend !== "") {
    result = JSON.parse(localStorage.getItem("history") || "[]");
    return result;
  }
  let res = await backend.get("/history");
  result = res.data.history as Array<TranslationHistory>;
  return result;
};
//This function will clear the history of translation
export const clearHistory = async () => {
  if (config.backend !== "") {
    localStorage.removeItem("history");
    return true;
  }
  let res = await backend.delete("/history");
  let result = res.data.result as boolean;
  return result;
};

File: flask-server/resources/history.py
from flask_restful import Resource
from flask import jsonify,session
from flask_jwt_extended import jwt_required
class History(Resource):
    @jwt_required()
    def get(self):
        return jsonify({'history': session.get('translation_history')})
    @jwt_required()
    def delete(self):
        session['translation_history'] = None
        return jsonify({'result': True})

    def add_translation_to_session(translation):
        if session.get('translation_history') != None:
            translation_list = session.get('translation_history')
            translation_list.insert(0, translation)
            session['translation_history'] = translation_list
        else:
            translation_list = []
            translation_list.append(translation)
            session['translation_history'] = translation_list
